### PR TITLE
Removed the deprecated code from axis.py

### PR DESCRIPTION
**Pull request by agautam478**
## PR summary
This PR removes the deprecated code during the version 3.8 from `lib/matplotlib/axis.py`  which is a sub issue in the parent issues here: https://github.com/matplotlib/matplotlib/issues/26865

## PR checklist

- [ N/A] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ N/A] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [N/A ] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [ N/A] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [N/A ] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Comment by ksunden** [comment]
https://matplotlib.org/stable/api/prev_api_changes/api_changes_3.8.0.html#tick-set-label1-and-tick-set-label2

Here is the release note from the deprecation

File: doc/api/next_api_changes/removals/26870-AG.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+``set_label1`` and ``set_label2``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (QuLogic) [inline doc/api/next_api_changes/removals/26870-AG.rst:2]: Should be more specific about location:
```suggestion
``matplotlib.axis.Tick.set_label1`` and ``matplotlib.axis.Tick.set_label2``
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Review by QuLogic — COMMENTED** [review]
The documentation build also says that `doc/api/axis_api.rst` also contains some old references to these removed items.

**Review by QuLogic — CHANGES_REQUESTED** [review]
Also marking as request-changes so that this isn't merged before 3.9 is out.

**Comment by agautam478** [comment]
pre-commit.ci autofix

**Comment by QuLogic** [comment]
Hi @agautam478; 3.9 is branched now, so this can go in. Do you mind rebasing so that the CI results will be current?

**Comment by ksunden** [comment]
Thank you for your effort, however these changes got rolled into a larger deprecation commit in  #28183, so closing now.

**Pull request closed without merging**